File: src/main/java/com/coffee/shop/api/demo/image/impl/ImageMapper.java
package com.coffee.shop.api.demo.image.impl;

import com.coffee.shop.api.demo.image.api.dto.ImageDto;
import com.coffee.shop.api.demo.image.api.model.Image;
import org.mapstruct.Mapper;

@Mapper(componentModel = "spring")
public interface ImageMapper {

    Image dtoToEntity(ImageDto dto);

    ImageDto entityToDto(Image entity);

}

File: src/main/kotlin/com/coffee/shop/api/demo/image/impl/ImageMapper.kt
package com.coffee.shop.api.demo.image.impl

import com.coffee.shop.api.demo.image.api.dto.ImageDto
import com.coffee.shop.api.demo.image.api.model.Image
import org.mapstruct.Mapper

@Mapper(componentModel = "spring")
interface ImageMapper {
    fun dtoToEntity(dto: ImageDto?): Image?
    fun entityToDto(entity: Image?): ImageDto?
}